src/installer.ts

```diff
@@ -436,17 +519,22 @@ function evaluateVersions(versions: string[], versionSpec: string): string {
 }
 
 export function getNodejsDistUrl(version: string) {
-  const prerelease = semver.prerelease(version);
-  if (version.includes('nightly')) {
-    return 'https://nodejs.org/download/nightly';
-  } else if (prerelease) {
-    return 'https://nodejs.org/download/rc';
+  switch (distributionOf(version)) {
+    case Distributions.CANARY:
+      core.debug('requested v8 canary distribution');
+      return 'https://nodejs.org/download/v8-canary';
+    case Distributions.NIGHTLY:
+      core.debug('requested nightly distribution');
+      return 'https://nodejs.org/download/nightly';
+    case Distributions.RC:
+      core.debug('requested release candidates distribution');
+      return 'https://nodejs.org/download/rc';
+    case Distributions.DEFAULT:
+      return 'https://nodejs.org/dist';
```

Comment:
I think we can remove debug because the function is called in three places. I think we should not repeat the log.